### PR TITLE
Initialize extension correctly

### PR DESCRIPTION
**Pull request by rgrinberg**
We should initialize the extension and register all commands even if the
language server fails to load.

### CODE REVIEW

**Review by mnxn — APPROVED** [review]
Looks good to me.
Thanks for the quick fix!

**Comment by rgrinberg** [comment]
@mnxn I pushed another commit to make things more robust. We should register the extension as soon as possible without waiting for any promises. Lastly, I've also simplified a bunch of redundant api's.

File: src/vscode_ocaml_platform.ml [hunk 1]
```diff
@@ -19,29 +19,42 @@ let activate (extension : ExtensionContext.t) =
...
-       sandbox: they may ignore the pop-up leaving the extension hanging, so we
-       use fallback; w/ a proper detection mechanism, we would redo work in rare
-       cases *)
-  then
-    let (_ : unit Promise.t) = suggest_to_pick_sandbox instance in
+  let sandbox = Sandbox.of_settings_or_detect () in
+  let (_ : unit Promise.t) =
+    let* sandbox = sandbox in
+    let is_fallback = Option.is_empty sandbox in
+    (* register things with vscode, making sure to register their disposables *)
```
Comment (mnxn) [inline src/vscode_ocaml_platform.ml:32]: I think this comment should be moved a couple lines up to where the registration functions are being called.

**Review by mnxn — APPROVED** [review]

File: src/vscode_ocaml_platform.ml (hunk 1 above)
Comment (rgrinberg) [reply src/vscode_ocaml_platform.ml]: I removed this comment entirely. It only explains the obvious.

**Pull request merged**